Support for IDEF instruction?

I'm looking at a font with VTT sources that use the `IDEF` instruction, and `vttLib.vtt_compile()` fails on that:

```
  File "xxx/venv/lib/python3.7/site-packages/vttLib/__init__.py", line 128, in split_functions
    assert 0, "Unexpected token in fpgm: %s" % t
AssertionError: Unexpected token in fpgm: ['IDEF', 145]
```

`IDEF` is very similar to `FDEF`, and I quickly tried a patch where everywhere it looks for `FDEF` it will also look for `IDEF`, but that must be too naive, and doesn't work.

Does anyone here have a hint (no pun intended) of how to implement this so I can try to make a PR after all? Although I'm afraid I might be over my head, because this is the first time I look at this code base and I don't know all that much about VTT to begin with.

--- a/src/vttLib/__init__.py
+++ b/src/vttLib/__init__.py
@@ -107,7 +107,7 @@ def split_functions(fpgm_tokens):
     tokens = iter(fpgm_tokens)
     for t in tokens:
         mnemonic = t.mnemonic
-        if mnemonic.startswith("FDEF"):
+        if mnemonic.startswith("FDEF") or mnemonic.startswith("IDEF"):
             body = [t]
             for t in tokens:
                 body.append(t)
@@ -127,6 +127,7 @@ def merge_functions(functions, include=None):
     for line in "\n".join(functions).splitlines():
         asm.extend(line.strip().split())
     funcs = {}
+    idefs = {}
     stack = []
     tokens = iter(asm)
     for token in tokens:
@@ -137,14 +138,18 @@ def merge_functions(functions, include=None):
                     stack.append(num)
                 except ValueError:
                     break
-        if token.startswith("FDEF"):
+        if token.startswith("FDEF") or token.startswith("IDEF"):
+            is_fdef = token.startswith("FDEF")
             num = stack.pop()
             body = [token]
             for token in tokens:
                 body.append(token)
                 if token.startswith("ENDF"):
                     break
-            funcs[num] = body
+            if is_fdef:
+                funcs[num] = body
+            else:
+                idefs[num] = body
             continue
         assert 0, "Unexpected token in fpgm: %s" % token
     if include is not None:
@@ -153,6 +158,10 @@ def merge_functions(functions, include=None):
     result = ["PUSH[]"] + [str(n) for n in sorted(funcs, reverse=True)]
     for num in sorted(funcs):
         result.extend(funcs[num])
+    result.append("PUSH[]")
+    result.extend([str(n) for n in sorted(idefs, reverse=True)])
+    for num in sorted(idefs):
+        result.extend(idefs[num])
     return result
 
 

--- a/src/vttLib/__init__.py
+++ b/src/vttLib/__init__.py
@@ -138,7 +138,7 @@ def merge_functions(functions, include=None):
                     stack.append(num)
                 except ValueError:
                     break
-        if token.startswith("FDEF") or token.startswith("IDEF"):
+        if token.startswith(("FDEF", "IDEF")):
             is_fdef = token.startswith("FDEF")
             num = stack.pop()
             body = [token]

--- a/tests/data/idef145.txt
+++ b/tests/data/idef145.txt
@@ -1,0 +1,6 @@
+/* define GETVARIATION in case this OS doesn't have it */
+IDEF[],145
+#BEGIN
+#PUSH,0, 0
+#END
+ENDF[]

--- a/tests/data/idef145_transformed.txt
+++ b/tests/data/idef145_transformed.txt
@@ -1,3 +1,5 @@
+PUSHB[ ]
+145
 IDEF[ ]	/* InstructionDefinition */
 PUSHB[ ]	/* 2 values pushed */
 0 0

--- a/tests/data/idef145_transformed.txt
+++ b/tests/data/idef145_transformed.txt
@@ -1,0 +1,4 @@
+IDEF[ ]	/* InstructionDefinition */
+PUSHB[ ]	/* 2 values pushed */
+0 0
+ENDF[ ]	/* EndFunctionDefinition */

--- a/tests/vttLib_test.py
+++ b/tests/vttLib_test.py
@@ -8,7 +8,7 @@ from vttLib import make_ft_program, pformat_tti, transform_assembly
 DATA_DIR = os.path.join(os.path.dirname(os.path.realpath(__file__)), "data")
 
 
-TEST_NAMES = ["fdef83", "fdef133", "fdef152", "fdef153"]
+TEST_NAMES = ["fdef83", "fdef133", "fdef152", "fdef153", "idef145"]
 
 
 @pytest.fixture(params=TEST_NAMES)

